tools/runner/logger.go

```diff
@@ -0,0 +1,98 @@
+package runner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+	"github.com/grpc/test-infra/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PodLogger provides functionality to save pod logs to files.
+type PodLogger struct {
+}
+
+// NewPodLogger creates a new PodLogger object.
+func NewPodLogger() *PodLogger {
+	return &PodLogger{}
+}
+
+// savePodLogs saves pod logs to files with same name as pod.
```

Comment:
I would make a distinction between functions that are part of the interface, such as savePodLogs, and functions that are purely internal. I would rename this function SavePodLogs for this reason. We could move the LogSaver into a subpackage, instead of keeping it in runner, and this would still be valid.